pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -185,104 +232,121 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 	// Start deploy klusterlet components
 	// Ensure the existence namespaces for klusterlet and klusterlet addon
 	// Sync pull secret to each namespace
-	namespaces := []string{config.KlusterletNamespace, fmt.Sprintf("%s-addon", config.KlusterletNamespace)}
-	for _, namespace := range namespaces {
-		err := n.ensureNamespace(ctx, klusterlet.Name, namespace)
-		if err != nil {
-			return err
-		}
-
-		// Sync pull secret
-		_, _, err = resourceapply.SyncSecret(
-			n.kubeClient.CoreV1(),
-			controllerContext.Recorder(),
-			n.operatorNamespace,
-			imagePullSecret,
-			namespace,
-			imagePullSecret,
-			[]metav1.OwnerReference{},
-		)
+	err = n.syncPullSecret(ctx, n.kubeClient, n.kubeClient, klusterlet.Name, config.KlusterletNamespace, controllerContext.Recorder())
+	if err != nil {
+		return err
+	}
+	// For now, whether in Default or Detached mode, the addons will be deployed on the managed cluster.
+	// sync image pull secret from management cluster to managed cluster for addon namespace
+	// TODO(zhujian7): In the future, we may consider deploy addons on the management cluster in Detached mode.
+	addonNamespace := fmt.Sprintf("%s-addon", config.KlusterletNamespace)
+	err = n.syncPullSecret(ctx, n.kubeClient, managedClusterClients.kubeClient, klusterlet.Name, addonNamespace, controllerContext.Recorder())
+	if err != nil {
+		return err
+	}
 
+	if detachedMode {
+		// In detached mode, we should ensure the namespace on the managed cluster since
+		// some resources(eg:service account) are still deployed on managed cluster.
+		err := n.ensureNamespace(ctx, managedClusterClients.kubeClient, klusterletName, config.KlusterletNamespace)
```

Comment:
should we still need this? OR why we only need to ensure this on detached mode?

Reply:
> why we only need to ensure this on detached mode?

In detached mode, the managed cluster and management cluster are separated, the namespace in management cluster has been ensured in the [syncPullSecret](https://github.com/open-cluster-management-io/registration-operator/pull/172/files#diff-276cfcd05be4cfe9f9381e2a2c7a64e51b09ee5e1c692e3121e7158347c6a166R501) func.